server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -325,15 +325,31 @@ export async function checkSoftDeleted(
 export async function executeApiWithMetric<U>(
 	api: () => Promise<U>,
 	apiName: string,
-	telemetryProperties: Record<string, any>,
+	apiCategory: string,
+	metricEnabled: boolean,
+	samplingPeriod?: number,
+	telemetryProperties?: Record<string, any>,
 ): Promise<U> {
+	// If generating a metric is not enabled, we just execute the API.
+	// We also do the same if sampling tells us to skip the metric for
+	// this instance (when a sampling period is provided).
+	if (!metricEnabled && samplingPeriod && getRandomInt(samplingPeriod) !== 0) {
```

Comment:
I think this should be `if (!metricEnabled || (samplingPeriod && getRandomInt(samplingPeriod) !== 0)`

Otherwise, it will always sample if `metricEnabled == false`

Reply:
Thank you for catching this! I didn't test this combination of values properly. I will update in the next iteration.

Reply:
Done 😄